packages/order-utils/src/eip712_utils.ts

```diff
@@ -100,18 +101,24 @@ export const eip712Utils = {
      * @param   txOrigin The desired `tx.origin` that should be able to submit an Ethereum txn involving this 0x transaction
      * @return  A typed data object
      */
-    createCoordinatorApprovalTypedData: (
+    async createCoordinatorApprovalTypedDataAsync(
         transaction: SignedZeroExTransaction,
         verifyingContract: string,
         txOrigin: string,
-    ): EIP712TypedData => {
+    ): Promise<EIP712TypedData> {
         const domain = {
             ...transaction.domain,
             name: constants.COORDINATOR_DOMAIN_NAME,
             version: constants.COORDINATOR_DOMAIN_VERSION,
             verifyingContract,
         };
-        const transactionHash = transactionHashUtils.getTransactionHashHex(transaction);
+        const transactionHash = await new DevUtilsContract('0x0000000000000000000000000000000000000000', {
```

Comment:
@feuGeneA instantiating a `DevUtilsContract` takes up quite a bit of memory because it instantiates an EVM instance under-the-hood (according to what @xianny has told me), so we only want to instantiate it once per app and pass it around.

In _this_ particular case, we should just move the TS implementation over the `contracts/coordinator` which is the only code using it, and _keep_ the TS implementation. It would live as a test utils in that package, which is fine.

Reply:
This goes for `eip712_utils` and `order_hash_utils`.

Reply:
As discussed on slack, this needs to stay the way it is for now, because other things are still using `eip712Utils`.